tests/causal_estimators/test_causalml_estimator.py

```diff
@@ -0,0 +1,216 @@
+import pytest
+import subprocess
+import sys
+
+from dowhy import CausalModel
+from dowhy.datasets import linear_dataset
+
+# To skip the tests if we cannot install causalml
+installed_failed = False
+# Hack to install causalml if not already present
+try:
+    __import__("causalml")
+except ImportError:
+    try:
+        subprocess.check_call([sys.executable, "-m", "pip", "install", "causalml"])
+        __import__("causalml")
+        # We import this later as we obtained this from causalml
+        from xgboost import XGBRegressor
+    except subprocess.CalledProcessError:
+        installed_failed = True
+
+class TestCausalMLEstimator:
+    '''
+        To test the basic functionality of the CauslML estimators
+    '''
+    
+    @pytest.mark.skipif(installed_failed)
+    def test_LRSRegressor(self):
+        # Defined a linear dataset with a given set of properties
+        data = linear_dataset(

```

Comment:
Can this dataset be a common fixture that you define once. This fixture can return a dataset, and then all methods can add the output of fixture's function as a parameter. 

See, e.g., the smtp_connection fixture [here](https://docs.pytest.org/en/latest/fixture.html).

This can help later when we may want to test for multiple datasets. Then only the fixture needs to be changed.